editing/edit-with-branch-versioning/src/main/java/com/esri/samples/edit_with_branch_versioning/EditWithBranchVersioningController.java

```diff
@@ -0,0 +1,352 @@
+          // if a feature is selected and the current branch version is not the default, update the feature's geometry
+          if(selectedFeature != null && !serviceGeodatabase.getVersionName().equals(defaultVersion)){
+            Point2D secondaryClickPoint = new Point2D(event.getX(), event.getY());
+            Point mapPoint = mapView.screenToLocation(secondaryClickPoint);
+            selectedFeature.setGeometry(mapPoint);
+            updateFeature(selectedFeature);
+          }
+        }
+      });
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Create a new branch version using user defined values
+   */
+  @FXML
+  private void handleCreateVersionButtonClicked() {
+
+    // validate version name input
+    if (name.getText().contains(".") || name.getText().contains(";") || name.getText().contains("'") ||
+      name.getText().contains("\"")) {
+      new Alert(Alert.AlertType.ERROR, "Please enter a valid version name").show();
+      return;
+    } else if (name.getText().length()> 0 && Character.isWhitespace(name.getText().charAt(0))) {
+      new Alert(Alert.AlertType.ERROR, "Version name cannot begin with a space").show();
+      return;
+    } else if (name.getText().length() > 62) {
+      new Alert(Alert.AlertType.ERROR, "Version name must not exceed 62 characters").show();
+      return;
+    } else if (name.getText().length() == 0) {
+      new Alert(Alert.AlertType.ERROR, "Please enter a version name").show();
+      return;
+    }
+
+    // validate version access input
+    if (accessComboBox.getSelectionModel().getSelectedItem() == null) {
+      new Alert(Alert.AlertType.ERROR, "Please select an access level").show();
+      return;
+    }
+
+    // set the user defined name, access level and description as service version parameters
+    ServiceVersionParameters newVersionParameters = new ServiceVersionParameters();
+    newVersionParameters.setName(name.getText());
+    newVersionParameters.setAccess(accessComboBox.getSelectionModel().getSelectedItem());
+    newVersionParameters.setDescription(description.getText());
+
+    // update the UI
+    createVersionButton.setText("Creating version....");
+    createVersionButton.setDisable(true);
+
+    // create a new version with the specified parameters
+    ListenableFuture<ServiceVersionInfo> createdVersion = serviceGeodatabase.createVersionAsync(newVersionParameters);
+    createdVersion.addDoneListener(() -> {
+      try {
+        // get the name of the created version and switch to it
+        ServiceVersionInfo createdVersionInfo = createdVersion.get();
+        userCreatedVersion = createdVersionInfo.getName();
+        switchVersion(userCreatedVersion);
+
+        // hide the form from the UI as the sample only allows 1 version to be created
+        createVersionVBox.setVisible(false);
+        switchVersionButton.setDisable(false);
+      } catch (Exception ex) {
```

Comment:
I think a newline here would help separate these two blocks a bit better
```suggestion
        switchVersionButton.setDisable(false);

      } catch (Exception ex) {
```